### ISSUE TITLE
Dynamic kubelet config metrics missing

### ISSUE THREAD

**Issue by odinuge** [kubernetes/kubernetes#83183]
**What happened**:
The kubelet metrics related to [Dynamic kubelet config](https://github.com/kubernetes/enhancements/issues/281), except for one, are missing from the kubelet metric endpoint.

This applies to these metrics:
```
kubelet_node_config_assigned
kubelet_node_config_active
kubelet_node_config_last_known_good
```

**What you expected to happen**:

I expect that the metrics mentioned above should be found in the `/metrics` endpoint on the kubelet.

**How to reproduce it (as minimally and precisely as possible)**:

Run kubelet with a valid `--dynamic-config-dir`, and `--feature-gates=DynamicKubeletConfig=true` (defaults to true).

or run the node e2e tests: `make test-e2e-node PARALLELISM=1 REMOTE=true TEST_ARGS='--feature-gates=DynamicKubeletConfig=true' FOCUS="DynamicKubeletConfig" SKIP=""`

**Anything else we need to know?**:

All e2e tests related to DynamicKubeletConfig fails in master (and in v1.16): https://testgrid.k8s.io/sig-node-kubelet#node-kubelet-serial

This bug/error was introduced in https://github.com/kubernetes/kubernetes/pull/81534

**Environment**:
- Kubernetes version (use `kubectl version`): `v1.16` && `master`
- Cloud provider or hardware configuration: `N/A`
- OS (e.g: `cat /etc/os-release`): `N/A`
- Kernel (e.g. `uname -a`): `N/A`
- Install tools: `N/A`
- Network plugin and version (if this is a network-related bug): `N/A`
- Others: `N/A`

**Comment by odinuge** [comment]
/assign

WIP proof of concept solution to fix (make it a bit easier to understand the problem) here: #83184

**Comment by zouyee** [comment]
@kubernetes/sig-node-test-failures

**Comment by liggitt** [comment]
it might be more appropriate to fix the 1.16 e2e tests run against skewed components to not rely on alpha-level metrics

**Comment by odinuge** [comment]
> @liggitt: it might be more appropriate to fix the 1.16 e2e tests run against skewed components to not rely on
> alpha-level metrics
> 
> edit: I misunderstood the issue. discussed with @logicalhan offline, and he has an idea to restore
> previous behavior

That sounds perfect! He answered in the PR, so hopefully we can get this working again. I have wrapped my head around this issue for quite a few hours now, so I understand that it is kinda hard to understand by just reading this issue. Here https://github.com/kubernetes/kubernetes/pull/83184#discussion_r328954605 is some more background info that hopefully makes sense.

**Comment by odinuge** [comment]
Looks like the CI jobs are cleaning up now!
![Screenshot from 2019-10-11 09-51-49](https://user-images.githubusercontent.com/1467188/66634215-e5c27580-ec0c-11e9-9902-10d111890823.png)

Finally down to single digit number test failures, and away from the 30+ :smile:
